connectors/sync_job_runner.py

```diff
@@ -215,16 +216,23 @@ async def _sync_done(self, sync_status, sync_error=None):
     @with_concurrency_control()
     async def sync_starts(self):
         if not await self.reload_connector():
-            return False
+            raise SyncJobStartError(f"Couldn't reload connector {self.connector_id}")
```

Comment:
I think it could be a generic error - we have similar logic in multiple places, for jobs for example. We can probably have similar error-handling with error named like `EntryNotFound` or similar.

Reply:
We do have `ConnectorNotFoundError` and `ConnectorJobNotFoundError` defined, and used in `check_job`:

https://github.com/elastic/connectors-python/blob/da8a867218d90e3523624dcc6a3c278bdb767dc9/connectors/sync_job_runner.py#L280-L285

But I think it makes sense to raise `SyncJobStartError` in method `sync_starts`?

Reply:
All depends on the catcher - if you want to distinguish errors and show different error messages - yes, otherwise probably good to keep only one